app/models/station.rb

```diff
@@ -0,0 +1,97 @@
+class Station < ActiveRecord::Base 
+
+  belongs_to :city
+  has_many :start_trips , class_name: "Trip", foreign_key: "start_station_id" do
+    def date_of_highest_number_of_trips
+      group(:start_date).order("count_start_date DESC").count(:start_date).first.first.to_s
+    end
+    def trip_count_of_highest_number_of_trips
+      group(:start_date).order("count_start_date DESC").count(:start_date).first.last
+    end
+    def bike_most_used_starting
+      group(:bike_id).order("count_bike_id DESC").count(:bike_id).first.first
+    end
+  end
+  has_many :end_trips , class_name: "Trip", foreign_key: "end_station_id"
+  
+  validates :name,
+            :dock_count,
+            :city_id,
+            :installation_date, presence: true
+
+  def self.write(station_details)
+    self.find_or_create_by(name: station_details[:name],
+                          lat: station_details[:lat],
+                          long: station_details[:long],
+                          dock_count: station_details[:dock_count],
+                          city_id: find_city_id(station_details[:city_name]),
+                          installation_date: station_details[:installation_date]
+                          )
+  end
+
+  def write_update(station, station_details)
+    station.update(name: station_details[:name],
+                          lat: station_details[:lat],
+                          long: station_details[:long],
+                          dock_count: station_details[:dock_count],
+                          city_id: find_city_id(station_details[:city_name]),
```

Comment:
rather than passing this `update` method a specific `city_id`, you can actually just pass it an entire `City` object if your associations are set up correctly. here, you could do `city: find_city(station_details[:city_name])`. it's preferable to pass objects around rather than specific data or id's